File: untracked_tests/test_eastmoney.js


async function test() {
    const url = "https://push2.eastmoney.com/api/qt/clist/get?cb=data&pn=1&pz=100&po=1&np=1&fltt=2&invt=2&fid=f62&fs=m:90+t:2+f:!50&fields=f12,f14,f2,f3,f62,f20";
    const resp = await fetch(url, {
        headers: {
            "User-Agent": "Mozilla/5.0",
            "Referer": "https://quote.eastmoney.com"
        }
    });
    const text = await resp.text();
    console.log(text);
}
test();

File: untracked_tests/test_eastmoney.go
package main

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"net/http"
)

func main() {
	url := "https://push2.eastmoney.com/api/qt/clist/get?pn=1&pz=100&po=1&np=1&fltt=2&invt=2&fid=f62&fs=m:90+t:2+f:!50&fields=f12,f14,f2,f3,f62,f20"
	req, _ := http.NewRequest("GET", url, nil)
	req.Header.Set("User-Agent", "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/119.0.0.0 Safari/537.36 Edg/119.0.0.0")
	resp, err := http.DefaultClient.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()
	body, _ := ioutil.ReadAll(resp.Body)
	
	var resData map[string]interface{}
	json.Unmarshal(body, &resData)
	out, _ := json.MarshalIndent(resData, "", "  ")
	fmt.Println(string(out))
}
